Assets/Scripts/Bullets/BulletController.cs

```diff
@@ -1,24 +1,39 @@
-﻿using UnityEngine;
+﻿using System.Collections;
+using UnityEngine;
 
 public class BulletController : MonoBehaviour
 {
+    [SerializeField]
+    private ParticleSystem explosionParticle;
+    [SerializeField]
+    private AudioSource bulletExplosionAudio;
+
+
+
     [SerializeField]
     private float fireSpeed = 20f;
 
 
+
     public void Fire(Transform fireTransform, GameObject tank)
     {
-        
+
         GameObject bullet = Instantiate(gameObject, fireTransform.position, fireTransform.rotation);
         Rigidbody bulletRigidbody = bullet.GetComponent<Rigidbody>();
 
         bulletRigidbody.velocity = fireSpeed * tank.transform.forward;
     }
 
-
+    
 
     private void OnTriggerEnter(Collider other)
     {
+        TankService.Instance.DestroyTankOrBullet(explosionParticle, bulletExplosionAudio);
```

Comment:
Move your logic to ExplosionController script and reuse that explosion controller for both tanks and bullets